src/Npgsql/NpgsqlConnection.cs

```diff
@@ -324,22 +324,23 @@ async Task OpenAsync(bool async, CancellationToken cancellationToken)
                 Debug.Assert(connector.Connection is null,
                     $"Connection for opened connector '{Connector?.Id.ToString() ?? "???"}' is bound to another connection");
 
-                ConnectorBindingScope = ConnectorBindingScope.Connection;
-                connector.Connection = this;
-                Connector = connector;
-
-                if (enlistToTransaction is not null)
-                    EnlistTransaction(enlistToTransaction);
-
-                timeout = new NpgsqlTimeout(connectionTimeout);
-
                 // Since this connector was last used, PostgreSQL types (e.g. enums) may have been added
                 // (and ReloadTypes() called), or global mappings may have changed by the user.
                 // Bring this up to date if needed.
                 // Note that in multiplexing execution, the pool-wide type mapper is used so no
                 // need to update the connector type mapper (this is why this is here).
                 if (connector.TypeMapper.ChangeCounter != TypeMapping.GlobalTypeMapper.Instance.ChangeCounter)
+                {
+                    timeout = new NpgsqlTimeout(connectionTimeout);
```

Comment:
nit: can just inline it.

Or on a 2nd look... I'm not sure why we're restarting the timeout... We may have already spent time in _pool.Get above - seems like we shouldn't restart ConnectionTimeout from scratch...

Reply:
Not exactly sure why I've added it...